log/sublogger.go

```diff
@@ -12,56 +11,37 @@ func NewSubLogger(name string) (*SubLogger, error) {
 		return nil, errEmptyLoggerName
 	}
 	name = strings.ToUpper(name)
-	RWM.RLock()
+	mu.Lock()
+	defer mu.Unlock()
 	if _, ok := SubLoggers[name]; ok {
-		RWM.RUnlock()
 		return nil, fmt.Errorf("'%v' %w", name, ErrSubLoggerAlreadyRegistered)
 	}
-	RWM.RUnlock()
 	return registerNewSubLogger(name), nil
 }
 
 // SetOutput overrides the default output with a new writer
-func (sl *SubLogger) SetOutput(o io.Writer) {
-	sl.mtx.Lock()
+func (sl *SubLogger) setOutput(o *multiWriterHolder) {
```

Comment:
If o is nil, I suggest using the default output that you've defined elsewhere:

```go
if o == nil {
	sl.output= getWriters(&SubLoggerConfig{
		Name:   strings.ToUpper(subLogger),
		Level:  "INFO|WARN|DEBUG|ERROR",
		Output: "stdout"})
}
```
Just prevents a panic. Or have an err return